main/main.go

```diff
@@ -227,136 +249,161 @@ func main() {
 	// Create listeners for each of the subnets configured as a source
 	errGroup, ctx := errgroup.WithContext(context.Background())
 	for _, s := range cfg.SourceBlockchains {
-		blockchainID, err := ids.FromString(s.BlockchainID)
-		if err != nil {
-			logger.Error(
-				"Invalid subnetID in configuration",
-				zap.Error(err),
-			)
-			panic(err)
-		}
 		sourceBlockchain := s
 
-		health := atomic.NewBool(true)
-		relayerHealth[blockchainID] = health
+		isHealthy := atomic.NewBool(true)
+		relayerHealth[s.GetBlockchainID()] = isHealthy
 
-		// errgroup will cancel the context when the first goroutine returns an error
 		errGroup.Go(func() error {
-			// Dial the eth client
-			ethClient, err := utils.DialWithConfig(
-				context.Background(),
-				sourceBlockchain.RPCEndpoint.BaseURL,
-				sourceBlockchain.RPCEndpoint.HTTPHeaders,
-				sourceBlockchain.RPCEndpoint.QueryParams,
-			)
-			if err != nil {
-				logger.Error(
-					"Failed to connect to node via RPC",
-					zap.String("blockchainID", sourceBlockchain.BlockchainID),
-					zap.Error(err),
-				)
-				return err
-			}
-
-			// Create the ApplicationRelayers
-			applicationRelayers, minHeight, err := createApplicationRelayers(
-				ctx,
-				logger,
-				metrics,
-				db,
-				ticker,
-				*sourceBlockchain,
-				network,
-				messageCreator,
-				&cfg,
-				ethClient,
-				destinationClients,
-			)
-			if err != nil {
-				logger.Error(
-					"Failed to create application relayers",
-					zap.String("blockchainID", sourceBlockchain.BlockchainID),
-					zap.Error(err),
-				)
-				return err
-			}
-			logger.Info(
-				"Created application relayers",
-				zap.String("blockchainID", sourceBlockchain.BlockchainID),
-			)
+			return relayer.ProcessManualWarpMessages(manualWarpMessages[sourceBlockchain.GetBlockchainID()])
```

Comment:
Sorry, I already removed the manual warp messages from the config. Do you think we should keep them in there for a few releases with a deprecation note? I can't imagine many people were using them already given the difficulty in having to add them to the nodes as well.

Reply:
Yeah, we should deprecate features gracefully at this point, even though I agree that likely few people are using this particular feature. Apologies for not making that clear up front

Reply:
So a couple thoughts here. We don't initialize the logger for the application until after we finish parsing the config. So it would be a little hard to emit logs from here. Returning an error and failing validation seems a little too extreme here, is there some middle ground?

Do we still want to process the manual warp messages if they're included in the config? It should be easy enough to point the users to the new API.